WikiDrips/WikiTableViewController.swift

```diff
@@ -52,13 +50,19 @@ class WikiTableViewController: UITableViewController, UISearchBarDelegate {
     }
     
     func search() {
-//      TODO: replace with async call to Wikipedia API
-        let someDateTime = Date(timeIntervalSinceNow: -(Double(wikiDocs.count) * 60))
-        let testData: [String: AnyObject] = ["title": "My new title" as AnyObject, "pubDate": someDateTime as AnyObject]
-        if let wikiDoc = WikiDoc(data: testData) {
-            wikiDocs.insert([wikiDoc], at: 0)
+        guard let searchText = searchText,
+            let wikiRequest = WikiRequest(searchText: searchText) else {
+            return
+        }
+
+        wikiRequest.fetchWikiDocs { newDocs in
+            DispatchQueue.main.async(){
+                if newDocs.count > 0 {
+                    self.wikiDocs = [newDocs]
```

Comment:
There are two ways to "fix" this:
1. Capture `[weak self]` instead:
```
        wikiRequest.fetchWikiDocs { [weak self] newDocs in
            DispatchQueue.main.async(){
                if newDocs.count > 0 {
                    self?.wikiDocs = [newDocs]
                    self?.tableView.reloadData()
                }
            }
        }
```
With method 1 then if the view controller gets deallocated by the time `self` is used in the block then self will be nil and nothing will get run. Also, by you not holding a strong reference to `self`, the view controller is free to be deallocated before the block finishes.

One disadvantage of method 1, is that self (i.e. the view controller) can go away at any point, so in theory its possible that self goes away right after `self?.wikiDocs = [newDocs]` is executed. In this particular case you might not care about that, but you can certainly imagine some code that you want the behavior to be: "I don't want to hold a strong reference to self before this block is executed, but if the block does start to be executed, I want the whole block to execute and not just part of it". In that scenario you want to do something like:

2. Capture a local strong `self`:
```
        wikiRequest.fetchWikiDocs { [weak self] newDocs in
            guard let strongSelf = self else { return }
            DispatchQueue.main.async(){
                if newDocs.count > 0 {
                    strongSelf.wikiDocs = [newDocs]
                    strongSelf.tableView.reloadData()
                }
            }
        }
```
This will only grab a strong reference to `self` if the `self` is still around when block starts.

The general pattern that Eldar and I follow is that for network call closures, we use [weak self] when there's only one line referencing self, and use strongSelf method when we have multiple lines.